ssc/cmod_wavefile.cpp

```diff
@@ -60,100 +78,255 @@ class cm_wave_file_reader : public compute_module
+        if (as_integer("use_specific_wf_wave") == 0)
+        {
+            getline(ifs, buf);
+            getline(ifs, buf1);
+
+            // header name value pairs
+            std::vector<std::string> keys = split(buf);
+            values = split(buf1);
+            int ncols = (int)keys.size();
+            int ncols1 = (int)values.size();
+
+            if (ncols != ncols1 || ncols < 13)
+            {
+                throw exec_error("wave_file_reader", "incorrect number of header columns: " + std::to_string(ncols));
+            }
+
+            assign("name", var_data(values[0]));
+            assign("city", var_data(values[1]));
+            assign("state", var_data(values[2]));
+            assign("country", var_data(values[3]));
+            // lat with S is negative
+            ssc_number_t dlat = std::numeric_limits<double>::quiet_NaN();
+            std::vector<std::string> slat = split(values[4], ' ');
+            if (slat.size() > 0)
+            {
+                dlat = std::stod(slat[0]);
+                if (slat.size() > 1)
+                {
+                    if (slat[1] == "S") dlat = 0.0 - dlat;
+                }
+            }
+            assign("lat", var_data(dlat));
+            // lon with W is negative
+            ssc_number_t dlon = std::numeric_limits<double>::quiet_NaN();
+            std::vector<std::string> slon = split(values[5], ' ');
+            if (slon.size() > 0)
+            {
+                dlon = std::stod(slon[0]);
+                if (slon.size() > 1)
+                {
+                    if (slon[1] == "W") dlon = 0.0 - dlon;
+                }
+            }
+            assign("lon", var_data(dlon));
+            assign("nearby_buoy_number", var_data(values[6]));
+            assign("average_power_flux", var_data(std::stod(values[7])));
+            assign("bathymetry", var_data(values[8]));
+            assign("sea_bed", var_data(values[9]));
+            assign("tz", var_data(std::stod(values[10])));
+            assign("data_source", var_data(values[11]));
+            assign("notes", var_data(values[12]));
+        }
+        // read in 21 rows x 22 columns
+
+        //ssc_number_t* mat = allocate("wave_resource_matrix", 21, 22);
+        //size_t numberRecords = 2920;
+        //ssc_number_t* wave_heights = allocate("wave_significant_height", numberRecords);
+        //ssc_number_t* wave_periods = allocate("wave_energy_period", numberRecords);
+        //if (values.size() != 22)
+        if (as_integer("wave_resource_model_choice") == 1)
+        {
+            size_t numberRecords = 0;
+            size_t year_index = 0, month_index = 0, day_index = 0, hour_index = 0, minute_index = 0, period_index = 0, height_index = 0;
+            getline(ifs, buf); //Skip past column labels for record counting
+            while (getline(ifs, buf))
+                numberRecords++;
+            assign("number_records", (int)numberRecords);
```

Comment:
is there an expected number of records? is subhourly ok? is a 8784 ok? Maybe do an error check here